scipy/ndimage/src/nd_image.c

```diff
@@ -297,24 +284,29 @@ static PyObject *Py_Correlate1D(PyObject *obj, PyObject *args)
 static PyObject *Py_Correlate(PyObject *obj, PyObject *args)
 {
     PyArrayObject *input = NULL, *output = NULL, *weights = NULL;
-    npy_intp *origin = NULL;
+    PyArray_Dims origin;
     int mode;
     double cval;
 
     if (!PyArg_ParseTuple(args, "O&O&O&idO&", NI_ObjectToInputArray, &input,
                           NI_ObjectToInputArray, &weights,
                           NI_ObjectToOutputArray, &output,
-                         &mode, &cval,
-                         NI_ObjectToLongSequence, &origin))
+                          &mode, &cval,
+                          PyArray_IntpConverter, &origin)) {
         goto exit;
+    }
+    if (!_validate_origin(input, origin)) {
+        goto exit;
+    }
     if (!NI_Correlate(input, weights, output, (NI_ExtendMode)mode, cval,
```

Comment:
NIce catch! There were actually many of those, have removed all I found in the latest commit.